auth/src/main/java/feast/auth/service/AuthorizationService.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.auth.service;
+
+import feast.auth.authorization.AuthorizationProvider;
+import feast.auth.authorization.AuthorizationResult;
+import feast.auth.config.SecurityProperties;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class AuthorizationService {
+
+  private final SecurityProperties securityProperties;
+  private final AuthorizationProvider authorizationProvider;
+
+  @Autowired
+  public AuthorizationService(
+      SecurityProperties securityProperties,
+      ObjectProvider<AuthorizationProvider> authorizationProvider) {
+    this.securityProperties = securityProperties;
+    this.authorizationProvider = authorizationProvider.getIfUnique();
+  }
+
+  /**
+   * Determine whether a user has access to the Project
+   *
+   * @param securityContext User's Spring Security Context. Used to identify user.
+   * @param project Name of the project for which membership should be tested.
+   */
+  public void authorizeRequest(SecurityContext securityContext, String project) {
+    Authentication authentication = securityContext.getAuthentication();
+    if (!this.securityProperties.getAuthorization().isEnabled()) {
+      return;
+    }
+
+    AuthorizationResult result = this.authorizationProvider.checkAccess(project, authentication);
```

Comment:
We will have to make sure that certificate lookup is cached in order to prevent latency spikes in serving.

Reply:
@woop I don't get what you mean? are you taking about caching across the requests?

Reply:
Yip, not exactly in scope for your PR. But the keys used to verify JWTs should be cached. There are libraries that make this easy. I think we should just ensure that there are no production impacts from moving authentication to serving.

Reply:
@woop I checked the flow in the code, `NimbusJwtDecoder.withJwkSetUri(jwkEndpointURI).build();` I do see caching in the library.